api/src/modules/scenario-interventions/scenario-interventions.service.ts

```diff
@@ -301,14 +257,33 @@ export class ScenarioInterventionsService extends AppBaseService<
    */
   async createSourcingLocationsDataForNewMaterialIntervention(
     dto: CreateScenarioInterventionDto,
-    sourcingData: SourcingLocationWithRecord[],
+    sourcingData: SourcingLocation[],
   ): Promise<SourcingData[]> {
-    const tonnage: number =
-      sourcingData.reduce(
-        (previous: any, current: any) => previous + Number(current.tonnage),
-        0,
-      ) *
-      (dto.percentage / 100);
+    const sourcingRecords: { year: number; tonnage: number }[] = [];
```

Comment:
Can you please add some minor comment to explain what it is happening here? For each existing element find the tonnage for the same year, if not exist add a new element to the array... Up to you how to explain it

Mostly to avoid our future selves to blow their brains